.github/actions/validate/action.yml

```
@@ -0,0 +1,26 @@
+name: Validate
+description: Check for dead links
+
+runs:
+  using: 'composite'
+  steps:
+    - uses: actions/checkout@v4
+    - uses: actions/setup-node@v4
+      with:
+        node-version: 20
```

```suggestion
        node-version: 20
        cache: npm
```

Might be worth trying?

https://github.com/actions/setup-node?tab=readme-ov-file#caching-global-packages-data